superset-frontend/src/hooks/useTruncation/useChildElementTruncation.ts

```diff
@@ -27,92 +27,65 @@ import { RefObject, useLayoutEffect, useState, useRef } from 'react';
+ * will be excluded from the hiddenElementCount.
+ * @param elementRef HTMLElement that will be checked to see if the sum of child element widths
+ * exceeds the visible clientWidth of the elementRef.
+ * @returns [isTruncated: boolean, hiddenElementCount: number]
+ */
+const useChildElementTruncation = (elementRef: RefObject<HTMLElement>) => {
+  const [isTruncated, setIsTruncated] = useState<boolean>(false);
+  const [hiddenElementCount, setHiddenElementCount] = useState<number>(0);
+
+  const { scrollWidth, clientWidth, childNodes } = elementRef?.current ?? {
     scrollWidth: 0,
-    parentElementWidth: 0,
-    plusRefWidth: 0,
-  });
+    clientWidth: 0,
+    childNodes: DEFAULT_ARRAY,
+  };
 
   useLayoutEffect(() => {
-    const currentElement = elementRef.current;
-    const plusRefElement = plusRef?.current;
-
-    if (!currentElement) {
-      return;
-    }
-
-    const { scrollWidth, clientWidth, childNodes } = currentElement;
-
-    // By using the result of this effect to truncate content
-    // we're effectively changing it's size.
-    // That will trigger another pass at this effect.
-    // Depending on the content elements width, that second rerender could
-    // yield a different truncate count, thus potentially leading to a
-    // rendering loop.
-    // There's only a need to recompute if the parent width or the width of
-    // the child nodes changes.
-    const previousEffectInfo = previousEffectInfoRef.current;
-    const parentElementWidth = currentElement.parentElement?.clientWidth || 0;
-    const plusRefWidth = plusRefElement?.offsetWidth || 0;
-    previousEffectInfoRef.current = {
-      scrollWidth,
-      parentElementWidth,
-      plusRefWidth,
-    };
-
-    if (
-      previousEffectInfo.parentElementWidth === parentElementWidth &&
-      previousEffectInfo.scrollWidth === scrollWidth &&
-      previousEffectInfo.plusRefWidth === plusRefWidth
-    ) {
-      return;
-    }
-
     if (scrollWidth > clientWidth) {
-      // "..." is around 6px wide
-      const truncationWidth = 6;
-      const plusSize = plusRefElement?.offsetWidth || 0;
-      const maxWidth = clientWidth - truncationWidth;
       const elementsCount = childNodes.length;
 
       let width = 0;
-      let hiddenElements = 0;
-      for (let i = 0; i < elementsCount; i += 1) {
-        const itemWidth = (childNodes[i] as HTMLElement).offsetWidth;
-        const remainingWidth = maxWidth - truncationWidth - width - plusSize;
-
-        // assures it shows +{number} only when the item is not visible
+      let hiddenElementCounter = 0;
+      childNodes.forEach((node: HTMLElement) => {
+        const itemWidth = node?.offsetWidth;
+        const remainingWidth = clientWidth - width;
+        /**
+         * Evaluate if hiddenElementCounter should increment before adding to width
+         * to ensure we do not count partially hidden / clipped / truncated element as a hidden item
+         */
```

Comment:
Just to make it similar to the other comments.
```suggestion
        // Evaluate if hiddenElementCounter should increment before adding to width
        // to ensure we do not count partially hidden / clipped / truncated element as a hidden item
```